deployer/helm_upgrade_decision.py

```diff
@@ -0,0 +1,358 @@
+"""
+Functions related to deciding which clusters and/or hubs need their *hub helm chart or
+support helm chart upgrading depending on an input list of filenames that have been
+added or modified in a GitHub Pull Request.
+"""
+import fnmatch
+import os
+from pathlib import Path
+from ruamel.yaml import YAML
+from rich.console import Console
+from rich.table import Table
+
+yaml = YAML(typ="safe", pure=True)
+
+# Determine if we are running a test or not. We set this env var to true in the
+# pytest.ini file so it is set when the package is tested.
+test_env = os.getenv("RUN_ENV", False)
+
+
+def discover_modified_common_files(modified_paths: list):
+    """There are certain common files which, if modified, we should upgrade all hubs
+    and/or all clusters appropriately. These common files include the helm charts we
+    deploy, as well as the GitHub Actions and deployer package we use to deploy them.
+    Args:
+        modified_paths (list[str]): The list of files that have been added or modified
+            in a given GitHub Pull Request.
+    Returns:
+        upgrade_all_clusters (bool): Whether or not all clusters should be upgraded
+            since the support chart has changed
```

Comment:
I'm not confident the logic of upgrading all clusters and helm charts in them makes sense just because we change the support chart we deploy on all/most clusters, but that is probably matching the current behavior we have so :+1: for now!